tests/ts_util.erl

```diff
@@ -522,30 +525,45 @@ get_optional(N, X) ->
     end.
 
 
--define(DELTA, 1.0e-15).
-
-assert_float(String, {_, {Cols, [ValsA]}} = Exp, {_, {Cols, [ValsB]}} = Got) ->
+assert_float(String, {ok, Thing1}, {ok, Thing2}) ->
+    assert_float(String, Thing1, Thing2);
+assert_float(String, {Cols, [ValsA]} = Exp, {Cols, [ValsB]} = Got) ->
     case assertf2(tuple_to_list(ValsA), tuple_to_list(ValsB)) of
-        fail -> lager:info("*****************", []),
+        fail ->
+            lager:info("*****************", []),
             lager:info("Test ~p failed", [String]),
             lager:info("Exp ~p", [Exp]),
             lager:info("Got ~p", [Got]),
             lager:info("*****************", []),
             fail;
-        pass -> pass
+        pass ->
+            pass
     end;
-assert_float(String, Exp, Got) -> assert(String, Exp, Got).
+assert_float(String, Exp, Got) ->
+    assert(String, Exp, Got).
 
 assertf2([], []) -> pass;
-assertf2([H1 | T1], [H2 | T2]) ->
-    Diff = H1 - H2,
-    Av = (H1 + H2)/2,
-    if Diff/Av > ?DELTA -> fail;
-        el/=se          -> assertf2(T1, T2)
-    end.
+assertf2([H1 | T1], [H2 | T2]) when is_float(H1), is_float(H2) ->
```

Comment:
The sign was obviously not taken into account. ~~When I fixed that (`Diff = abs(H1-H2)`), I still had false negatives with values like `1788996260097066800.0` vs `1.788996260097067e18`, and decided to compare the strings, which worked.~~

After double-checking, it appears the abs fix is enough (the false negative was due to me taking some shortcuts in development process). I am reverting this commit now.